src/main/java/net/minestom/server/entity/EntityProjectile.java

```diff
@@ -111,26 +150,22 @@ public void tick(long time) {
             setNoGravity(true);
             onStuck();
         } else {
-            if (!super.onGround) {
-                return;
-            }
-            super.onGround = false;
-            setNoGravity(false);
-            onUnstuck();
+            onHit(((State.HitEntity) state).entity);
+            remove();
```

Comment:
I think whether the projectile is removed here should depend on which projectile it is. For example, it doesn't apply to fishing hooks (which should stay inside the entity) and piercing arrows (which should go through the entity). Subclasses can call `remove()` in `onHit(Entity)` if needed.